main.rb

```diff
@@ -0,0 +1,173 @@
+    end
+    puts 'Thank you for using our Library!'
+  end
+
+  def menu
+    puts 'Please choose an option by entering a number'
+    puts '1 - List all books'
+    puts '2 - List all people'
+    puts '3 - Create a person'
+    puts '4 - Create a book'
+    puts '5 - Create a rental'
+    puts '6 - List all rentals for a given person ID'
+    puts '7 - Exit app'
+  end
+
+  def get_num(option)
+    case option
+    when '1'
+      list_all_books
+    when '2'
+      list_all_people
+    when '3'
+      create_person
+    when '4'
+      create_book
+    when '5'
+      create_rental
+    when '6'
+      list_rentals_p_id
+    else
+      puts 'Please enter a number between 1 and 7'
+    end
+  end
+
+  def list_all_books
+    puts 'There are no books listed yet' if @books.empty?
+
+    @books.each { |book| puts "Title: #{book.title}, Author: #{book.author}" }
+    sleep 1.2
+  end
+
+  def list_all_people
+    puts 'There are no people listed yet' if @people.empty?
+
+    @people.each { |person| puts "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" }
+    sleep 1.2
+  end
+
+  def create_person
+    print 'Enter (1) to create a student and (2) to create a teacher: '
+    option = gets.chomp
+
+    case option
+    when '1'
+      create_student
+    when '2'
+      create_teacher
+    else
+      puts 'Input invalid. Please enter (1) to create a student and (2) to create a teacher: '
+    end
+  end
+
+  def create_student
+    print 'Name: '
+    name = gets.chomp.capitalize
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Has parent permission? [Y/N]: '
+    parent_permission = gets.chomp.downcase
+
+    student = Student.new(age, name, parent_permission)
+    @people << student
+
+    puts 'Student created'
+    sleep 1.2
+  end
+
+  def create_teacher
+    print 'Name: '
+    name = gets.chomp.capitalize
+
+    print 'Age: '
+    age = gets.chomp.to_i
+
+    print 'Specialization: '
+    specialization = gets.chomp
+
+    teacher = Teacher.new(specialization, age, name)
+    @people << teacher
+
+    puts 'Teacher created'
+    sleep 1.2
+  end
+
+  def create_book
+    print 'Book Title: '
+    title = gets.chomp.capitalize
+
+    print 'Author: '
+    author = gets.chomp.capitalize
+
+    book = Book.new(title, author)
+    @books << book
+
+    puts 'Book added'
+    sleep 1.2
+  end
+end
+
+def create_rental
+  puts 'Select one of the books of the list below: '
+  @books.each_with_index { |book, index| puts "#{index}) Title: #{book.title}, Author: #{book.author}" }
+
+  book_id = gets.chomp.to_i
+
+  puts 'Select a person number (not id) from the list below'
+  @people.each_with_index do |person, index|
+    puts "#{index}) [#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}"
+  end
+
+  person_id = gets.chomp.to_i
+
+  print 'Date: '
+  date = gets.chomp.to_s
+
+  rental = Rental.new(date, @people[person_id], @book[book_id])
+  @rentals << rental
```

Comment:
- [ ] when creating a rental the user receives the below error due to the array in `@people` being empty/not accessible.

![Screenshot 2022-01-28 093048](https://user-images.githubusercontent.com/39240820/151513566-d9cfc838-b7a6-4051-9926-928b92e02448.png)